onedal/linear_model/incremental_linear_model.py

```diff
@@ -0,0 +1,150 @@
+# ==============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import numpy as np
+
+from daal4py.sklearn._utils import get_dtype
+
+from ..common.hyperparameters import get_hyperparameters
+from ..datatypes import _convert_to_supported, from_table, to_table
+from ..utils import _check_X_y, _num_features
+from .linear_model import BaseLinearRegression
+
+
+class IncrementalLinearRegression(BaseLinearRegression):
+    """
+    Incremental Linear Regression oneDAL implementation.
+
+    Parameters
+    ----------
+    fit_intercept : bool, default=True
+        Whether to calculate the intercept for this model. If set
+        to False, no intercept will be used in calculations
+        (i.e. data is expected to be centered).
+
+    copy_X : bool, default=True
+        If True, X will be copied; else, it may be overwritten.
+
+    algorithm : string, default="norm_eq"
+        Algorithm used for computation on oneDAL side
+    """
+
+    def __init__(self, fit_intercept=True, copy_X=False, algorithm="norm_eq"):
+        module = self._get_backend("linear_model", "regression")
```

Comment:
Warning is required since this class is experimental and subject to change.

```suggestion
    def __init__(self, fit_intercept=True, copy_X=False, algorithm="norm_eq"):
        warn("IncrementalLinearRegression is experimental and might be changed or removed in future.")
        module = self._get_backend("linear_model", "regression")
```

Reply:
I have no preferrable way. Let's choose one of them and finally merge this PR.